app/bundles/EmailBundle/Swiftmailer/Transport/AmazonTransport.php

```diff
@@ -60,23 +85,388 @@ class AmazonTransport extends \Swift_SmtpTransport implements CallbackTransportI
     /**
      * AmazonTransport constructor.
      *
-     * @param string              $host
      * @param Http                $httpClient
      * @param LoggerInterface     $logger
      * @param TranslatorInterface $translator
      * @param TransportCallback   $transportCallback
      */
-    public function __construct($host, Http $httpClient, LoggerInterface $logger, TranslatorInterface $translator, TransportCallback $transportCallback)
+    public function __construct(Http $httpClient, LoggerInterface $logger, TranslatorInterface $translator, TransportCallback $transportCallback)
     {
-        parent::__construct($host, 2587, 'tls');
-        $this->setAuthMode('login');
-
         $this->logger            = $logger;
         $this->translator        = $translator;
         $this->httpClient        = $httpClient;
         $this->transportCallback = $transportCallback;
     }
 
+    /**
+     * @return mixed
+     */
+    public function getRegion()
+    {
+        return $this->region;
+    }
+
+    /**
+     * @param string $region
+     */
+    public function setRegion($region)
+    {
+        $this->region = $region;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getUsername()
+    {
+        return $this->username;
+    }
+
+    /**
+     * @param $username
+     */
+    public function setUsername($username)
+    {
+        $this->username = $username;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getPassword()
+    {
+        return $this->password;
+    }
+
+    /**
+     * @param $password
+     */
+    public function setPassword($password)
+    {
+        $this->password = $password;
+    }
+
+    /**
+     * SES authorization and choice of region
+     * Initializing of TokenBucket.
+     *
+     * @return SesClient
+     *
+     * @throws \Exception
+     */
+    public function start()
+    {
+        $client = new SesClient([
+            'credentials' => new Credentials(
+                $this->getUsername(),
+                $this->getPassword()
+            ),
+            'region'  => $this->getRegion(),
+            'version' => 'latest',
```

Comment:
Are you sure this will work in the future? I think there should be a proper version which is tested and works.